### ISSUE TITLE
always_put_control_body_on_new_line conflicts with dart formatter

### ISSUE THREAD

**Issue by dvorapa** [dart-lang/linter#1847]
**Describe the issue**
dart formatter puts control body on new line only when line is too long, but linter always throws this warning. Thus dart formatter creates linter warnings.

**To Reproduce**
```
$ cd myapp
$ flutter format .
$ flutter analyze
   info • Separate the control structure expression from its statement • lib/main.dart:1446:23 • always_put_control_body_on_new_line
   info • Separate the control structure expression from its statement • lib/main.dart:1448:33 • always_put_control_body_on_new_line
```

**Expected behavior**
 Both tools should be in sync.

**Additional context**
Latest stable Flutter v1.9.1+h6 with Dart 2.5.0

**Comment by dvorapa** [comment]
Another option would be to mention in docs (like we do for lines_longer_than_80_chars), that 99 % is done by dartfmt, but 1 % is still left to a user

**Comment by pq** [comment]
I'm all for a doc improvement.

This one is used as is in the Flutter repo:

https://github.com/flutter/flutter/blob/cd052a56cba67bbc96fdf9a3ef940c3b3ca17a55/analysis_options.yaml#L48

and does reflect Flutter style.

**Comment by nailgilaziev** [comment]
any workarounds exist for keeping **linter** and **formatter** consistent to each other? 
except commenting `always_put_control_body_on_new_line`

**Comment by a14n** [comment]
`always_put_control_body_on_new_line` was design for Flutter codebase that don't use `dartfmt`. You can disable this lint in the analysis_options.yaml if you use dartfmt. A workaround could be to always use curly braces for control flow bodies.

**Comment by nailgilaziev** [comment]
@a14n , sorry, I didn't understand your answer, about that Flutter doesn't use `dartfmt`. Can you clarify that point?
I meet this problem in AndroidStudio in Flutter project in simple function (not a widget builder function)
My simple function and widget building functions live together in one file.

ps moreover you can meet this in builder function (no meaning code attached)
![image](https://user-images.githubusercontent.com/2749021/74197068-b29dfa80-4c6f-11ea-9663-6e8614fddcb2.png)

**Comment by pq** [comment]
The lint was designed to help the Flutter team specifically with their formatting conventions. Unfortunately they diverge from what the Dart formatter does and so if you want to use it, you'll want to disable this lint.

**Comment by srawlins** [comment]
Do we have a "Only for the flutter SDK engineers" badge we can apply to this rule?

**Comment by bwilkerson** [comment]
No, but we have a way of documenting when a lint conflicts with one or more other lints, so we might want to document when they conflict with the formatter.

**Comment by srawlins** [comment]
Ah yeah. That's a good plan.
